compiler/semtypes.nim

```diff
@@ -252,8 +252,12 @@ proc semRangeAux(c: PContext, n: PNode, prev: PType): PType =
     else:
       result.n.add semConstExpr(c, range[i])
 
-  if (result.n[0].kind in {nkFloatLit..nkFloat64Lit} and classify(result.n[0].floatVal) == fcNan) or
-      (result.n[1].kind in {nkFloatLit..nkFloat64Lit} and classify(result.n[1].floatVal) == fcNan):
+  when not declared(isNaN):
+    static: doAssert (NimMajor, NimMinor, NimPatch) < (1, 5, 1) # sanity check to ensure this workaround isn't taken in recent nim
```

Comment:
```suggestion
```

removing this sanity check because of this bug: https://github.com/nim-lang/Nim/issues/16646